pkg/workload/schemachange/generate_test.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package schemachange
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPickOne(t *testing.T) {
+	r := rand.New(rand.NewSource(0))
+
+	v, err := PickOne(r, []int{0, 1, 2, 3, 4})
+	require.NoError(t, err)
+	require.Equal(t, 4, v)
+
+	_, err = PickOne(r, ([]int)(nil))
+	require.ErrorIs(t, err, ErrCaseNotPossible)
+}
+
+func TestPickAtLeast(t *testing.T) {
+	r := rand.New(rand.NewSource(0))
+
+	require.Panics(t, func() {
+		_, _ = PickAtLeast(r, -1, []int{})
+	})
+
+	out, err := PickAtLeast(r, 0, []int{})
+	require.NoError(t, err)
+	require.Equal(t, []int{}, out)
+
+	_, err = PickAtLeast(r, 1, []int{})
+	require.ErrorIs(t, err, ErrCaseNotPossible)
+
+	out, err = PickAtLeast(r, 3, []int{1, 1, 1})
+	require.NoError(t, err)
+	require.Equal(t, []int{1, 1, 1}, out)
+}
+
+func TestPickBetween(t *testing.T) {
```

Comment:
nit: A general question: for some inputs (e.g. atLeast < 0 or atLeast > atMost), we panic, and for some other inputs (e.g. atLeast=1 but array is empty), we return `ErrCaseNotPossible` error. Why such a distinction? Is it bc the former is considered as "invalid input" and should not happen at all, while the latter is "valid but null input" and will happen during operation generation in the workload?

Reply:
Exactly that. Incorrectly picking `atLeast` or `atMost` is considered a programming error as those values should be controlled. On the other hand, the slice is a "set of possibilities" that is dependent on external factors.